### PR TITLE
[GH1781] .NET Core 2 support

### PR DESCRIPTION
**Pull request by adamhathcock**
https://github.com/cake-build/cake/issues/1781

Tests and Cake.exe all target - net461;netcoreapp2.0
Cake DLLs all target - netstandard2.0

Updated/adjustment references where necessary.

Notes:
- [x] Unit test with Encoding not working on `net461` and `netcoreapp2.0` but works on `netcoreapp1.0` (bug report filed https://github.com/dotnet/corefx/issues/26544 )
- [x] Need to figure out how to run integration tests.
- [x] Adjust building for `netcoreapp2.0`
- [x] Adjust packaging for `netcoreapp2.0`
- [x] Adjust integration tests for `netcoreapp2.0`

Looking for:
- Notes if this is the right path
- What else to needs looking at.

### CODE REVIEW

**Comment by devlead** [comment]
Perhaps this work will sort it?
https://github.com/cake-build/cake/pull/1778

**Comment by adamhathcock** [comment]
Ill see if I can merge that.  I did the package updates ignored the warnings.  That work should sort that.

Not sure if that will help my problem tests but we’ll see.

**Comment by ErikSchierboom** [comment]
@adamhathcock #1778 should fix the xUnit tests not running on .NET Core 2.0.

**Comment by adamhathcock** [comment]
The tests do work for me.  There's only two specifically that don't.

**Comment by ErikSchierboom** [comment]
@adamhathcock Which are those? On my branch all tests work.

**Comment by adamhathcock** [comment]
Should_Transform_Xml_String_And_Xsl_String_To_Result_String_With_Utf32Xml_Declaration and Should_Add_References_To_Session

I haven't looked into detail but my first guess is that you're not actually executing the tests on `netcoreapp2.0` but on `netcoreapp1.0` The Xunit updates allow the .NET Core 2 SDK to run tests on either runtime which is still good.

Hopefully, today I'll take your work and merge it to see what happens.

**Comment by adamhathcock** [comment]
Merged into a branch locally.  Gets rid of all the xunit warnings :)

I still get this error for `netcoreapp2.0` and `net461` (under mono) but not under `netcoreapp1.0` testing:
```
    Cake.Common.Tests.Unit.XML.XmlTransformationTests+TheTransformMethod.Should_Transform_Xml_String_And_Xsl_String_To_Result_String_With_Utf32Xml_Declaration [FAIL]
      Assert.Equal() Failure
                ↓ (pos 0)
      Expected: <?xml version="1.0" encoding="utf-32"?>
      Actual:   ﻿<?xml version="1.0" encoding="utf-32"?
                ↑ (pos 0)
```

Some encoding thing that needs to be figured out I guess.  The other failing test I just need to adjust compile flags.

**Comment by ErikSchierboom** [comment]
@adamhathcock FYI, my xUnit PR has just been [merged](https://github.com/cake-build/cake/commit/a40d5c455b9115a620d8c2117dea377ae6641ea7).

**Comment by adamhathcock** [comment]
@ErikSchierboom I know :)  I've already merged develop into my branch.  Looking at integration tests and nuget packaging (I think there should only be one package)

File: build.cake [hunk 1]
```diff
@@ -192,9 +203,13 @@ Task("Zip-Files")
...
-    // .NET Core
-    var coreclrFiles = GetFiles( parameters.Paths.Directories.ArtifactsBinNetCore.FullPath + "/**/*");
-    Zip(parameters.Paths.Directories.ArtifactsBinNetCore, parameters.Paths.Files.ZipArtifactPathCoreClr, coreclrFiles);
+    // .NET Core 1.0
+    var coreclrFiles1 = GetFiles( parameters.Paths.Directories.ArtifactsBinNetCore1.FullPath + "/**/*");
+    Zip(parameters.Paths.Directories.ArtifactsBinNetCore1, parameters.Paths.Files.ZipArtifactPathCoreClr, coreclrFiles1);
+    
+    // .NET Core 2.0
+    var coreclrFiles2 = GetFiles( parameters.Paths.Directories.ArtifactsBinNetCore2.FullPath + "/**/*");
+    Zip(parameters.Paths.Directories.ArtifactsBinNetCore2, parameters.Paths.Files.ZipArtifactPathCoreClr, coreclrFiles2);
```
Comment (ErikSchierboom) [inline build.cake]: I see that the .NET Core 1.0 and 2.0 files are combined in a single zip-file, is that correct?

**Comment by devlead** [comment]
Yes, if it doesn't blow up too big proper tools folder with framework moniker folders should be the way to go.

File: build.cake (hunk 1 above)
Comment (adamhathcock) [reply build.cake]: Nope!

Comment (ErikSchierboom) [reply build.cake]: Okay, I misread then :)

Comment (devlead) [reply build.cake]: Yip zip files should be separate, nuget if possible combined.

**Comment by adamhathcock** [comment]
Now building just one nuget.  Of course it's 15 megs vs 5 megs per moniker.  Even though it's big, I'd rather just have one.

**Comment by devlead** [comment]
agreed

**Comment by adamhathcock** [comment]
Currently stuck on this integration test:
```
Installing NuGet package xunit.assert...
Executing: /Users/adam/git/personal/cake/tests/integration/tools/nuget.exe install "xunit.assert" -OutputDirectory "/Users/adam/git/personal/cake/tests/integration/tools/Addins/xunit.assert.2.3.0-beta5-build3769" -Version "2.3.0-beta5-build3769" -Prerelease -ExcludeVersion -NonInteractive
Error: System.ComponentModel.Win32Exception (0x80004005): Operation not permitted
   at Interop.Sys.ForkAndExecProcess(String filename, String[] argv, String[] envp, String cwd, Boolean redirectStdin, Boolean redirectStdout, Boolean redirectStderr, Int32& lpChildPid, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd)
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start()
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at Cake.Core.IO.ProcessRunner.Start(FilePath filePath, ProcessSettings settings) in /Users/adam/git/personal/cake/src/Cake.Core/IO/ProcessRunner.cs:line 89
   at Cake.NuGet.NuGetPackageInstaller.InstallPackage(PackageReference package, DirectoryPath path) in /Users/adam/git/personal/cake/src/Cake.NuGet/NuGetPackageInstaller.cs:line 183
   at Cake.NuGet.NuGetPackageInstaller.Install(PackageReference package, PackageType type, DirectoryPath path) in /Users/adam/git/personal/cake/src/Cake.NuGet/NuGetPackageInstaller.cs:line129
   at Cake.Core.Scripting.ScriptProcessor.InstallAddins(IReadOnlyCollection`1 addins, DirectoryPath installPath) in /Users/adam/git/personal/cake/src/Cake.Core/Scripting/ScriptProcessor.cs:line 93
   at Cake.Core.Scripting.ScriptRunner.Run(IScriptHost host, FilePath scriptPath, IDictionary`2 arguments) in /Users/adam/git/personal/cake/src/Cake.Core/Scripting/ScriptRunner.cs:line 152
   at Cake.Commands.BuildCommand.Execute(CakeOptions options) in /Users/adam/git/personal/cake/src/Cake/Commands/BuildCommand.cs:line 35
   at Cake.CakeApplication.Run(CakeOptions options) in /Users/adam/git/personal/cake/src/Cake/CakeApplication.cs:line 45
   at Cake.Program.Main() in /Users/adam/git/personal/cake/src/Cake/Program.cs:line 74
```

I guess `nuget.exe` still has to be a dependency?

**Comment by adamhathcock** [comment]
I guess I'll look into seeing why https://github.com/cake-build/cake/pull/1768 isn't being used here.

**Comment by devlead** [comment]
@adamhathcock it's opt in at the moment, i.e. via command line ` --nuget_useinprocessclient=true`

**Comment by adamhathcock** [comment]
I didn’t see that.  So much to do :(

I’ll go over the compile flags much more closely.

This would be so much easier without mono worries.

**Comment by gep13** [comment]
@adamhathcock what mono worries?

**Comment by mholo65** [comment]
yeah, we'd probably want to resolve that on runtime? If it's not possible, leave it as `netstandard1.6` but then we can't load `netstandard2.0` addins.

**Comment by adamhathcock** [comment]
Basically, the above.  Managing which runtime to execute on.

I think I’m just moaning that managing entry points is hard and I’d like to remove more things to make my life easier.

Just ignore me :) I don’t really have anything concrete to suggest.

**Comment by adamhathcock** [comment]
Hah, new problem.

The net461 tests seem to want to use .net standard 2.0 libraries.  This messes up the runtime detection as it's mostly compile flag based.  This means running under `net461` tries to use `netstandard2.0` librares.  It thinks it's .net core 2.0 :(  There's no proper API https://github.com/dotnet/corefx/issues/17452

I guess I can introduce some reflection.

**Comment by ErikSchierboom** [comment]
@adamhathcock Any updates?

**Comment by adamhathcock** [comment]
I was going down the route of breaking changes, I'm afraid.

The current issue with the tests is that it uses build time flags (as does CakeRuntime) to know what runtime is currently executing.  In the .NET Standard 2.0 world, this won't fly.  It actually wasn't too robust before.  A .NET Standard 2.0 library can be run on `net461` and higher and `netcoreapp2.0` This breaks a lot of the assumptions in a few key areas of Cake.

I was creating a new `Runtime` enum that is similar to the `TestRuntime` enum that would replace `CakeRuntime.TargetFramework` (which only had two possible values).  The new values for `Runtime` would be Clr, CoreClr and CoreClr2.

With these flags, the tests would better know what it's being ran on instead of using build time flags.

I was going to do some more to prove it out but I don't see a way around it with out breaking changes.

I'm going pretty slow between work and a terrible personal life at the moment :(

**Comment by patriksvensson** [comment]
@adamhathcock No worries. First make sure that you take care of yourself and the personal life. Cake is always there, so no rush with the PR :)

You're raising a good point with the `ICakeRuntime` stuff that we need to find some solution to. The architecture behind that choice was made pre .NET Core 1.0, so many assumptions about things might be wrong now.

@cake-build/cake-team Any input on this?

**Comment by ErikSchierboom** [comment]
@adamhathcock Breaking changes may be needed, as you are basically supporting an entire new platform. Good luck with your personal situation!

**Comment by ErikSchierboom** [comment]
@adamhathcock @patriksvensson Is there any plan on how to proceed with this PR?

**Comment by adamhathcock** [comment]
Sorry everyone.  Between work taking me further from .NET land and my personal life, I don't think I'll be able to complete this soon.

I hope someone will fork or start over using something from this PR to get Cake running natively on .NET Core 2.

I saw @seanfarrow asking about it.  I'm happy to help transfer knowledge.

**Comment by SeanFarrow** [comment]
Hey @AdamHathcock,

I’m happy to help, is there any chance we could have a Skype to understand what you have done?
Feel free to ping me privately, my Email is:
Sean.farrow@seanfarrow.co.uk
Cheers
Sean.

**Comment by patriksvensson** [comment]
I also think it would be good if someone from the Cake team could participate in this as well since it affects a lot of internals.

**Comment by SeanFarrow** [comment]
I agree.

**Comment by adamhathcock** [comment]
I can do mornings (like 6am) or evenings like (7 pm or later) UK time.  Maybe need to organize this somewhere than GitHub.

**Comment by SeanFarrow** [comment]
Hi, yes, evening would be better for me.
Where are you based?

**Comment by adamhathcock** [comment]
Job has me going through Manchester/Leeds/Sheffield which is why I don’t have free time 😬

**Comment by SeanFarrow** [comment]
Nice, I feel for you! I used to do that a lot, are weekends any good for anyone?

**Comment by adamhathcock** [comment]
I can only do weekdays really.  I’m adam.hathcock on Skype.  Just installed it again.

**Comment by SeanFarrow** [comment]
Ok, do one of the cake team wanto come along? If yes, who and what times work?

**Comment by adamhathcock** [comment]
Just add me and we'll start a group chat to organize

**Comment by patriksvensson** [comment]
@adamhathcock @SeanFarrow I created a Gitter room where we can start the conversation and decide on a time for a chat

**Comment by SeanFarrow** [comment]
What’s the room address?

**Comment by adamhathcock** [comment]
Back on the case.

**Comment by devlead** [comment]
Nice 👍

**Comment by adamhathcock** [comment]
Current stuck issue on `netcoreapp1.0` tests only

```
   System.TypeInitializationException : The type initializer for 'NSubstitute.Core.SubstitutionContext' threw an exception.
      ---- System.IO.FileNotFoundException : Could not load file or assembly 'System.Diagnostics.TraceSource, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find thefile specified.
```

Blergh!

**Comment by adamhathcock** [comment]
Well, using the cake script results in that.  Running from the command line manually doesn't do that.  Pain

**Comment by devlead** [comment]
@adamhathcock grace got rebased against latest developments so is not a SDK issue?

**Comment by adamhathcock** [comment]
I merged in the latest `develop`  I've updated xunit to fix other issues.  

Not using the `dotnet-xunit` tool seems to fix that issue.  I'm not sure what the difference is.

**Comment by patriksvensson** [comment]
@adamhathcock Does it still throw if you remove `-noshadow` from dotnet-xunit?

**Comment by adamhathcock** [comment]
@patriksvensson yeah...I run `dotnet xunit` directly in the Cake.Common.Tests directory and get the issue.

**Comment by adamhathcock** [comment]
Working through NuGet framework resolution errors.  Not sure what to be aiming for with these tests:
```
Failed   Cake.NuGet.Tests.Unit.NuGetContentResolverTests+Addins.Should_Return_Exact_Framework_If_Possible(framework: ".NETStandard,Version=v1.6", expected: "netstandard1.6")
Error Message:
 Assert.Equal() Failure
                          ↓ (pos 16)
Expected: /Working/lib/netstandard1.6/file.dll
Actual:   /Working/lib/net461/file.dll
                          ↑ (pos 16)
Stack Trace:
   at Cake.NuGet.Tests.Unit.NuGetContentResolverTests.Addins.Should_Return_Exact_Framework_If_Possible(String framework, String expected) in /Users/adam/git/misc/cake/src/Cake.NuGet.Tests/Unit/NuGetContentResolverTests.cs:line 152
Failed   Cake.NuGet.Tests.Unit.NuGetContentResolverTests+Addins.Should_Return_Exact_Framework_Even_Though_Files_Located_In_Root(framework: ".NETStandard,Version=v1.6", expected: "netstandard1.6")
Error Message:
 Assert.Equal() Failure
                          ↓ (pos 16)
Expected: /Working/lib/netstandard1.6/file.dll
Actual:   /Working/lib/net461/file.dll
                          ↑ (pos 16)
Stack Trace:
   at Cake.NuGet.Tests.Unit.NuGetContentResolverTests.Addins.Should_Return_Exact_Framework_Even_Though_Files_Located_In_Root(String framework, String expected) in /Users/adam/git/misc/cake/src/Cake.NuGet.Tests/Unit/NuGetContentResolverTests.cs:line 269
Failed   Cake.NuGet.Tests.Unit.NuGetContentResolverTests+Addins.Should_Return_Nearest_Compatible_Framework_If_An_Exact_Match_Is_Not_Possible(framework: ".NETStandard,Version=v1.6", expected: "netstandard1.5")
Error Message:
 Assert.Equal() Failure
                          ↓ (pos 16)
Expected: /Working/lib/netstandard1.5/file.dll
Actual:   /Working/lib/net452/file.dll
                          ↑ (pos 16)
Stack Trace:
   at Cake.NuGet.Tests.Unit.NuGetContentResolverTests.Addins.Should_Return_Nearest_Compatible_Framework_If_An_Exact_Match_Is_Not_Possible(String framework, String expected) in /Users/adam/git/misc/cake/src/Cake.NuGet.Tests/Unit/NuGetContentResolverTests.cs:line 173
```

**Comment by adamhathcock** [comment]
Nevermind, figured it out.  My new `Runtime` enum has to be used :)

**Comment by SeanFarrow** [comment]
I was just looking at the ScriptRunnerTests.TheRunMethod.Should_Add_References_To_Session failing test to see if I can help fix it.
Is this failing due to the fact that the test is broken, in that we look for one assembly rather than 2 in certain cases, or is it a problem with the code in some way?
@adamhathcock, if you could give me some pointers, I'm happy to help where I can as I need this for a project asap, even if it's unreleased.

**Comment by adamhathcock** [comment]
@SeanFarrow honestly, I'm unsure about those.  I'll have to look again when I get time and just beat my head against it.

**Comment by SeanFarrow** [comment]
Is it meant to add one or two references, we can make the test pass if we need to.

**Comment by adamhathcock** [comment]
I posted this in the gitter room:

If there's more people to add to this channel that would be great. I've gotten the unit tests to pass but everything else around the build and integration tests is very difficult. I'm going to need help.
It might be better to have a feature branch on the main repo to get more people involved?

**Comment by SeanFarrow** [comment]
Ok, which room was this?

**Comment by SeanFarrow** [comment]
That room doesn’t seem to exist, for me at least!

**Comment by patriksvensson** [comment]
@SeanFarrow I've added you to the room

**Comment by SeanFarrow** [comment]
Hi,

I still can’t see this, what’s the uri? With the one below, I get a 404.

File: src/Cake.Core/Cake.Core.csproj [hunk 2]
```diff
@@ -1,41 +1,20 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
-
   <PropertyGroup>
     <AssemblyName>Cake.Core</AssemblyName>
-    <TargetFrameworks>net46;netstandard1.6</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0</TargetFrameworks>
```
Comment (mholo65) [inline src/Cake.Core/Cake.Core.csproj]: We'd still need `net46` until OmniSharp moves to `net461` or `net47`.

**Comment by ErikSchierboom** [comment]
@adamhathcock Just out of curiosity: how are things going with this?

**Comment by adamhathcock** [comment]
@ErikSchierboom I'm not aware of any lingering issues when I run things locally.  Things look to be build issues but I can only view appveyor

The appveyor issue was building nuget packages but it works for me locally (macOS).

I will update the branch again from master though.

I'll also rebase and do another round of cleanup when things are looking better

**Comment by ErikSchierboom** [comment]
Great to hear! Good luck.

**Comment by devlead** [comment]
@adamhathcock so I took this for a spin, sorry for not doing so earlier! Great work!
I've in my fork
* Squashed the 38 commits to 1 (rebase was a hassle otherwise 😎 )
* Rebased against latest develop
* Added back net46 as per @mholo65 feedback and reverted a few ifdefs because that.
* Added a xnunit workaround for Mono so tests don't blow up on MacOS / Linux
* Upgraded to .NET Core SDK 2.1.4 which has the latest security fixes
* Added NoRestore options added in Cake 0.25.0 for faster builds
* Updated Cake bitrise docker container to use .NET Core SDK 2.1.4

Submitted PR #2012 just to test CI:
image

There's still likely some quirks(complaints about duplicate tests), but it builds & tests succeed on Win, Mac and Linux.

@adamhathcock With your blessing I would like to force push my changes to your fork so it goes into this #1812 PR and we can start iron out the last kinks from there. Please take a look at #2012 first so I haven't missed any obvious.

I haven't inspected the integration tests yet, but they do pass on Windows & Mac for me, but we can test more once this #1812 PR is up to date.

**Comment by devlead** [comment]
> Adam Hathcock @adamhathcock 18:03
> @devlead Sounds great! Push away! I’ll work any last kinks I can see.

git push -f  ✅

File: src/Cake.Common.Tests/Unit/XML/XmlTransformationTests.cs [hunk 3]
```diff
@@ -182,7 +182,7 @@ public void Should_Transform_Xml_String_And_Xsl_String_To_Result_String_With_Xml
                 Assert.Equal(htm, result, ignoreLineEndingDifferences: true);
             }
 
-            //this feels wrong to be CLR only
+            // this feels wrong to be CLR only
```
Comment (devlead) [inline src/Cake.Common.Tests/Unit/XML/XmlTransformationTests.cs]: Yes, could this be an .NET Core issue we you raise upstream? Looks like a BOM is received, even if  it's set to false or something with GetString(). Haven't gone in detail, but it does feel weird.

**Comment by adamhathcock** [comment]
I made an issue for the encoding thing.  I'm not sure what else needs to be done for this PR.

I'd still like better detection of runtime and build times but looking at examples from other projects, things aren't much better: https://github.com/dotnet/BenchmarkDotNet/blob/master/src/BenchmarkDotNet.Core/Portability/RuntimeInformation.cs

Suggestions for other things that might need work or clean up are open!

**Comment by devlead** [comment]
@adamhathcock great work 👍If it indeed is a framework bug ( dotnet/corefx#26544 ), then disabling the test for .NET Core, like you've end -  is basically what we can do,a long with adding an issue to track it so we can re-enable it once it's fixed and released.

Meanwhile could you rebase and squash into one commit,  and then @cake-build/cake-team can start taking it for a spin before merging.

**Comment by adamhathcock** [comment]
Squashed down.  Ready to beat it up again https://github.com/orgs/cake-build/teams/cake-team

**Comment by adamhathcock** [comment]
@patriksvensson No worries!  I enjoy consistency.  Hopefully, the periods are up to your satisfaction now :)

File: src/Cake.Core/Polyfill/AssemblyHelper.cs [hunk 4]
```diff
@@ -34,26 +34,20 @@ public static Assembly LoadAssembly(AssemblyName assemblyName)
...
             {
                 throw new ArgumentNullException(nameof(path));
             }
 
             if (path.Segments.Length == 1 && !fileSystem.Exist(path))
             {
-                // Not a valid path. Try loading it by its name.
-                return Assembly.Load(new AssemblyName(path.FullPath));
+               // Not a valid path. Try loading it by its name.
+               return Assembly.Load(new AssemblyName(path.FullPath));
```
Comment (mholo65) [inline src/Cake.Core/Polyfill/AssemblyHelper.cs]: Wrong indentation?

File: src/Cake.Core/Polyfill/EnvironmentHelper.cs [hunk 5]
```diff
@@ -31,17 +31,35 @@ public static bool Is64BitOperativeSystem()
         public static PlatformFamily GetPlatformFamily()
         {
 #if NETCORE
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            try
```
Comment (mholo65) [inline src/Cake.Core/Polyfill/EnvironmentHelper.cs]: Why was these `try`-`catch` introduced? At what point will we see `PlatformNotSupportedException`?

File: src/Cake.NuGet.Tests/Cake.NuGet.Tests.csproj [hunk 6]
```diff
@@ -1,44 +1,35 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
-
   <PropertyGroup>
     <AssemblyName>Cake.NuGet.Tests</AssemblyName>
-    <TargetFrameworks>net46;netcoreapp1.0</TargetFrameworks>
-    <PlatformTarget>AnyCpu</PlatformTarget>
+    <TargetFrameworks>net461;netcoreapp2.0</TargetFrameworks>
```
Comment (mholo65) [inline src/Cake.NuGet.Tests/Cake.NuGet.Tests.csproj]: Why not `net46` as Cake.NuGet targets `net46` and `netstandard2.0`?

File: src/Cake.Core/ICakeRuntime.cs [hunk 7]
```diff
@@ -13,10 +14,15 @@ namespace Cake.Core
...
+        /// Gets the build-time .NET framework version that is being used.
+        /// </summary>
+        FrameworkName BuiltFramework { get; }
+
+        /// <summary>
+        /// Gets the current executing .NET Runtime.
         /// </summary>
         /// <returns>The target framework.</returns>
-        FrameworkName TargetFramework { get; }
+        Runtime Runtime { get; }
```
Comment (mholo65) [inline src/Cake.Core/ICakeRuntime.cs]: Breaking changes... We sould also bump the `breaking change` version [here](https://github.com/cake-build/cake/blob/develop/src/Cake.Core/Constants.cs#L11)

File: src/Cake.Testing.Xunit/Cake.Testing.Xunit.csproj [hunk 8]
```diff
@@ -1,31 +1,18 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
-
   <PropertyGroup>
     <AssemblyName>Cake.Testing.Xunit</AssemblyName>
-    <TargetFrameworks>net46;netstandard1.6</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0</TargetFrameworks>
```
Comment (mholo65) [inline src/Cake.Testing.Xunit/Cake.Testing.Xunit.csproj]: Should this also target `net46`?

File: src/Cake.Common.Tests/Unit/XML/XmlTransformationTests.cs [hunk 9]
```diff
@@ -182,7 +182,8 @@ public void Should_Transform_Xml_String_And_Xsl_String_To_Result_String_With_Xml
                 Assert.Equal(htm, result, ignoreLineEndingDifferences: true);
             }
 
-            [Fact]
+            // this feels wrong to be CLR only
+            [RuntimeFact(TestRuntime.Clr)]
```
[the remark this answers is not in the document]
Comment (devlead) [reply src/Cake.Common.Tests/Unit/XML/XmlTransformationTests.cs]: Because dotnet/corefx#26544

**Comment by mholo65** [comment]
Debugging (.NET Framework and .NET Core) + Intellisense seems to work fine. 👍

File: src/Cake.Common.Tests/Unit/XML/XmlTransformationTests.cs (hunk 9 above)
[the remark this answers is not in the document]
Comment (mholo65) [reply src/Cake.Common.Tests/Unit/XML/XmlTransformationTests.cs]: doh.

**Comment by devlead** [comment]
@adamhathcock ping us when you've had a chance to look at the last feedback given by @mholo65, and we'll then do a final review :shipit:

File: src/Cake.Core/Polyfill/EnvironmentHelper.cs (hunk 5 above)
Comment (adamhathcock) [reply src/Cake.Core/Polyfill/EnvironmentHelper.cs]: Not all platforms support this kind of probing.  I saw this in another code base that was doing the same thing to try to determine the environment.

File: src/Cake.Common.Tests/Cake.Common.Tests.csproj [hunk 10]
```diff
@@ -1,57 +1,45 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
-
```
[the remark this answers is not in the document]
Comment (adamhathcock) [reply src/Cake.Common.Tests/Cake.Common.Tests.csproj]: I'd rather leave it consistent with no empty lines.

**Comment by adamhathcock** [comment]
In general, I don't think there should be a `net46` target and should let downstream projects move up to the `net461` service release which is also `netstandard2.0` but that's not my call.  I'm fixing the targetting and some formatting things.

File: build.cake [hunk 11]
```diff
@@ -172,19 +189,20 @@ Task("Copy-Files")
...
         Configuration = parameters.Configuration,
         OutputDirectory = parameters.Paths.Directories.ArtifactsBinNetCore,
         MSBuildSettings = msBuildSettings
     });
 
     // Copy license
-    CopyFileToDirectory("./LICENSE", parameters.Paths.Directories.ArtifactsBinFullFx);
-    CopyFileToDirectory("./LICENSE", parameters.Paths.Directories.ArtifactsBinNetCore);
+    CopyFileToDirectory("./LICENSE", parameters.Paths.Directories.ArtifactsBin + "/net461");
+    CopyFileToDirectory("./LICENSE", parameters.Paths.Directories.ArtifactsBin + "/netcoreapp2.0");
```
[the remark this answers is not in the document]
Comment (adamhathcock) [reply build.cake]: It was necessary when I added a third target but that changed.  Will revert

**Comment by mholo65** [comment]
@adamhathcock, unfortuneately there must be a `net46` target for intellisense to work today. `OmniSharp.exe` loads `Cake.Core.dll` from the cake path in order to get the Script Host object type.

**Comment by adamhathcock** [comment]
I've read through https://github.com/OmniSharp/omnisharp-roslyn/pull/915 and it feels depressing.  Oh well.

**Comment by adamhathcock** [comment]
There's still some issue because `NuGet.PackageManagement.NetStandard` targets a netstandard 1 but not 2.  It's probably not a real problem but it's annoying to see package downgrade notices.

**Comment by adamhathcock** [comment]
Updated and commits flattened

**Comment by adamhathcock** [comment]
Updated the Constants.cs to 0.26

File: src/Cake.Core/ICakeRuntime.cs (hunk 7 above)
Comment (adamhathcock) [reply src/Cake.Core/ICakeRuntime.cs]: Fixed

**Comment by joskoanicic** [comment]
When this gets pulled is there going to be a new release shortly after or will you wait for some other changes to get into the next release? Just evaluating if I should wait for the official release. (Sorry if this is not the place to ask this question, not looking to spam).

**Comment by devlead** [comment]
@joskoanicic yes, next release will be this PR.

**Comment by cgibbons** [comment]
Any indication on when this will be merged and a release cut?  I'm trying to find a build tool that meets my needs, i.e. builds in a docker container whether windows or linux with no mono or full net framework dependencies.

**Comment by mholo65** [comment]
@adamhathcock thank you for your contribution! :+1: Awesome work!

**Comment by ErikSchierboom** [comment]
Thanks a lot @adamhathcock! Looking for the next release.

**Comment by gep13** [comment]
@adamhathcock thanks again for doing this work!  Really great work! 😄

**Comment by SeanFarrow** [comment]
I couldn’t agree more.

Do we have any idea when a release with this in will ship?
I’m currently using mono within my docker files and would like to fully switch.
Cheers

**Comment by gep13** [comment]
@SeanFarrow soon.

I know that isn't the best answer, but we are just confirming a few things, prior to the release.

**Comment by bonesoul** [comment]
any updates on this?

**Comment by devlead** [comment]
@bonesoul merged and shipped a several months ago.

**Comment by bonesoul** [comment]
sorry i guess i missed stuff on documentation.

still couldn't find how to get started with coreclr version.

**Pull request merged**